plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt

```diff
@@ -601,29 +602,48 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
           .anchor(focalPoint)
           .build(),
         actionAfter = {
-          cameraAnimationsPlugin.anchor = scaleCachedAnchor
-          notifyOnScaleListeners(detector)
-          spanSinceLast = abs(detector.currentSpan - detector.previousSpan)
+          onScaleAnimationEnd(detector)
         }
       )
     } else {
       val zoomBy =
         ln(detector.scaleFactor.toDouble()) / ln(PI / 2) * ZOOM_RATE.toDouble() * internalSettings.zoomAnimationAmount.toDouble()
-      easeToImmediately(
-        CameraOptions.Builder()
-          .zoom(mapCameraManagerDelegate.cameraState.zoom + zoomBy)
-          .anchor(focalPoint)
-          .build(),
-        actionAfter = {
-          cameraAnimationsPlugin.anchor = scaleCachedAnchor
-          notifyOnScaleListeners(detector)
-          spanSinceLast = abs(detector.currentSpan - detector.previousSpan)
+      if (internalSettings.simultaneousRotateAndPinchToZoomEnabled) {
+        val zoom = cameraAnimationsPlugin.createZoomAnimator(
+          cameraAnimatorOptions(mapCameraManagerDelegate.cameraState.zoom + zoomBy) {
+            startValue(mapCameraManagerDelegate.cameraState.zoom)
+            owner(MapAnimationOwnerRegistry.GESTURES)
+          }
+        ) {
+          duration = 0
         }
-      )
+        zoom.addListener(
+          onEnd = {
+            onScaleAnimationEnd(detector)
+          }
+        )
+        cameraAnimationsPlugin.playAnimatorsSequentially(zoom)
+      } else {
+        easeToImmediately(
+          CameraOptions.Builder()
+            .zoom(mapCameraManagerDelegate.cameraState.zoom + zoomBy)
+            .anchor(focalPoint)
+            .build(),
+          actionAfter = {
+            onScaleAnimationEnd(detector)
+          }
+        )
+      }
     }
     return true
   }
 
+  private fun onScaleAnimationEnd(detector: StandardScaleGestureDetector) {
+    cameraAnimationsPlugin.anchor = scaleCachedAnchor
```

Comment:
looking at code now I wonder - is setting anchor needed when `simultaneousRotateAndPinchToZoomEnabled == true`? we don't even change it then in `zoom` animator

Reply:
This is extracted from the original code, I'm not sure why we need to reset anchor.